packages/api-reference/src/hooks/index.ts

```diff
@@ -1,7 +1,8 @@
 export * from './useDarkModeState'
-export * from './useNavigation'
+export * from './useNavigate'
```

Comment:
@hanspagel @amritk I realise this is an existing pattern but it is one we should be looking to factor out. Usually with hooks the approach would be to have a single function export (like `useNavigate`) and that export would just expose the internal methods of that module. So we would do:
```
const  { getHeadingId, navigate  } = useNavigate()
```
Which is a pretty nice pattern for a few reasons:
- Strong association between a method and it's logical module
- Ability to have only a single import and access methods as needed either namespaced in a single variable or destructured into individual methods
- Ability to instantiate state at a module level or inside a hook instance depending on situation

Alternatively we can use ES module style import and then just import the functions directly from hooks/useNavigate. Exporting all functions from all hooks into a single index.ts file is just a recipe for naming conflicts and confusion of which hook a function belongs to. I would not generally call these hooks as this is really just a semi-isolated module with exports